test/x3/linker_error_test/config.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef LINKER_ERROR_EXAMPLE_CONFIG_HPP
```

Comment:
Same here. Please conform to the coding style; copyright at the top. Also the please follow the indentation style of the rest of the codebase.

Reply:
I could copy the ../rule1.cpp copyright, but really the code source is mentioned in, as I said
above, in main.cpp.  I guess I'll put that guys name in the copyright.

Reply:
Boost requires it. Actually, there's a check on commit on such things, IIRC.

Reply:
I'll put a copyright with my name in *all* the linker_error_test files.  
Since lukas didn't copyright it, I guess that's OK.

Reply:
I am not sure it is ok because you cannot just put a licence on code you did not write. Also this test is over-complicated IMO.

Reply:
Good point.  I'll simplify and them putting my copyright on it would be OK.

Reply:
An attempt was made to simplify the test here:

[complex_test](https://github.com/cppljevans/spirit-experiments/tree/BOOST_SPIRIT_INSTANTIATE/test/x3/linker_error_test)

starting with a prototype of such a simplification here:

[simple_test](https://github.com/cppljevans/spirit-experiments/tree/get_rhs/workbench/x3/BOOST_SPIRIT_INSTANTIATE_link_error)

Unfortunately, the link error didn't occur with the
simplified test, although trace output shown here:

[simple_out](https://github.com/cppljevans/spirit-experiments/blob/get_rhs/workbench/x3/BOOST_SPIRIT_INSTANTIATE_link_error/make_compare.out)

did demonstrate that with the patch, only 1 call to
`rule_definition<...>::parse` occured, but without the patch
(IOW using the existing spirit code), 2 calls to that parse
function occurred, suggesting (very strongly) that with the
existing spirit code, there is a redundant call to
`rule_definition<...>::parse`, and this is probably causing
the linke error *in some, as of yet, not understood,
circumstances*.

So, I see 2 choices:

* take the time to understand why there's no link
error with the simplified test and, with that understanding,
design a newer "a little less" simplified test that
reproduces the link error which was the initial reason for
this patch.

* just leave the simplified test code as is and replace the
  existing somewhat complicated code, which reproduces the
  link error, with the simplified test, which doesn't.

So, what should I do?

Reply:
I'm with choice 1: we need to get to the bottom of this "as of yet, not understood,
circumstances." :-)